File: Task3/Task3.swift
func sumAndProductOfDigits(_ n: Int) -> (Int, Int) {
    var sum = 0
    var product = 1
    var num = n
    while num > 0 {
        let digit = num % 10
        sum += digit
        product *= digit
        num /= 10
    }
    return (sum, product)
}

print("Кол-во чисел: ", terminator: "")
let n = Int(readLine()!)!
var indices = ""

for i in 0..<n {
    print("\(i) число: ", terminator: "")
    let num = Int(readLine()!)!
    let (sum, product) = sumAndProductOfDigits(num)
    if sum < product {
        indices += "\(i) "
    }
}

print(indices)

File: Task3/Task3.kt
fun sumAndProductOfDigits(n: Int): Pair<Int, Int> {
    var sum = 0
    var product = 1
    var num = n
    while (num > 0) {
        val digit = num % 10
        sum += digit
        product *= digit
        num /= 10
    }
    return Pair(sum, product)
}

fun main() {
    println("Кол-во чисел: ")
    val n = readLine()!!.toInt()
    val indices = mutableListOf<Int>()
    
    for (i in 0 until n) {
        println("$i число: ")
        val num = readLine()!!.toInt()
        val (sum, product) = sumAndProductOfDigits(num)
        if (sum < product) {
            indices.add(i)
        }
    }
    println(indices.joinToString(" "))
}